src/test/groovy/AppUserServiceImplTest.groovy

```diff
@@ -0,0 +1,106 @@
+import com.example.indiproject.DTOs.LoginDTO
+import com.example.indiproject.models.AppUser
+import com.example.indiproject.repositories.AppUserRepository
+import com.example.indiproject.services.api.AppUserService
+import com.example.indiproject.services.impl.AppUserServiceImpl
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class AppUserServiceImplTest extends Specification{
+    AppUserRepository appUserRepository = Mock()
+
+    @Subject
+    AppUserService appUserService = new AppUserServiceImpl(appUserRepository)
+
+    @Unroll
+    def "Should check if the user exists"() {
+        given:
+        appUserRepository.findByUsername("Kuba007") >> new AppUser("Kuba007","123",0.5);
+        when:
+        def result = appUserService.appUserWithUserNameExists(name);
+        then:
+        result == expectedResult
+        where:
+        name || expectedResult
+        "Kuba007" || true
```

Comment:
Myslim ze tady staci i jeden znak "pipe". A ackoliv to neni nutne, jsme zvykli parametry uvadene ve `where` davat i do signatury metody, nejak jako

```groovy
void "Should check if the user exists"(String name, boolean expectedResult) {
  ...
```